src/lavinmq/mqtt/sparkplug/state_tracker.cr

```diff
@@ -0,0 +1,119 @@
+module LavinMQ
+  module MQTT
+    module Sparkplug
+      record EdgeNodeState,
+        last_birth_timestamp : Int64,
+        last_death_timestamp : Int64? do
+        # Edge node is online if it has never died, or if the last birth is more recent than last death
+        def online? : Bool
+          death_ts = last_death_timestamp
+          death_ts.nil? || last_birth_timestamp > death_ts
+        end
+      end
+
+      class StateTracker
```

Comment:
I guess `StateTracker` will be used to more than `#track_birth` and `#track_death`? If not it has a lot of dead weight.